dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordPropertyMapping.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Runtime.InteropServices;
+using Microsoft.SemanticKernel.Data;
+
+namespace Microsoft.SemanticKernel.Connectors.Sqlite;
+
+/// <summary>
+/// Contains helper methods with property mapping for SQLite.
+/// </summary>
+internal static class SqliteVectorStoreRecordPropertyMapping
+{
+    public static string MapVector<TVector>(TVector vector)
+    {
+        if (vector is ReadOnlyMemory<float> memoryFloat)
+        {
+            return $"[{string.Join(", ", memoryFloat.ToArray())}]";
+        }
+
+        throw new NotSupportedException($"Mapping for type {typeof(TVector).FullName} is not supported.");
```

Comment:
Out of context it may be hard to understand that this is in the context of a vector, if e.g. someone added an int the wrong field and all they get is that we don't support mapping ints.  Consider:
```suggestion
        throw new NotSupportedException($"Mapping of type '{typeof(TVector).FullName}' to a vector is not supported.");
```